### PR TITLE
Feature/observability ddb

### PR DESCRIPTION
**Pull request by 96malhar**
## Description

This PR adds observability to to DynamoDB operations is DynamoDB HLL by creating traces using the SDK observability APIs.
* Traces are only created for public APIs that make a network call.
* I have updated the Async public APIs to await the `Task` instead of returning it. This is done to ensure that the trace is not disposed before the function completes its execution.

## Motivation and Context

This change is required as part of the Observability initiative for the SDK and HLLs
DOTNET-7679

## Testing

* All existing Unit and Integrations tests pass locally
* **Successful dry-run** Build Id: 6777f6d5-722a-4ca1-b141-39336ba73fe6

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: generator/.DevConfigs/9903b621-ba47-4a5a-92ec-1bfbfadb70f7.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "services": [
+        {
+            "serviceName": "DynamoDBv2",
+            "type": "minor",
+            "changeLogMessages": [
+                "Added the ability to create observability traces during DynamoDB operations"
```
Comment (normj) [inline generator/.DevConfigs/9903b621-ba47-4a5a-92ec-1bfbfadb70f7.json:7]: To make sure it is clear this is for the HLL how about something like
```
Added support for creating observability traces from the DataModel and DocumentModel abstractions for DynamoDB
```

**Review by normj — APPROVED** [review]
Approved but please update the dev config change lot message to include that this is specifically for the high level libraries.

**Review by philasmar — APPROVED** [review]

File: generator/.DevConfigs/9903b621-ba47-4a5a-92ec-1bfbfadb70f7.json (hunk 1 above)
Comment (96malhar) [reply generator/.DevConfigs/9903b621-ba47-4a5a-92ec-1bfbfadb70f7.json]: Done

**Comment by 96malhar** [comment]
> Approved but please update the dev config change lot message to include that this is specifically for the high level libraries.

Update the changelog message to what you suggested

**Pull request merged**